packages/lsp/src/connection.ts

```diff
@@ -268,9 +268,11 @@ export class LSPConnection extends LspWsConnection implements ILSPConnection {
     if (this.isDisposed) {
       return;
     }
-    Object.values(this.serverRequests).forEach(request =>
-      request.clearHandler()
-    );
+    if (this.serverRequests) { /*if statement to check if serverrequest is empty*/
```

Comment:
Looks good! I would adjust the comment, something like:

```suggestion
    if (this.serverRequests) {    // `serverRequests` may be undefined if dispose is called during initialization sequence
```
To avoid this:

![image](https://github.com/user-attachments/assets/5b1fc10b-af1b-4950-8b39-74ed84f31dab)

:)